pgjdbc/src/main/java/org/postgresql/jdbc/TimestampUtils.java

```diff
@@ -183,7 +153,8 @@ private Calendar getCalendar(int sign, int hr, int min, int sec) {
     int second = 0;
     int nanos = 0;
 
-    @Nullable Calendar tz = null;
+    boolean hasOffset = false;
```

Comment:
this was added to get away with null/non-null checks. For legacy APIs where calendar is used, this is the signal to use default calendar